### PR TITLE
Verify phar file

### PR DESCRIPTION
**Pull request by keradus**
Closing #3004

### CODE REVIEW

**Comment by keradus** [comment]
hm... nightly is failing :( (unrelated to my changes IMO)

**Comment by SpacePossum** [comment]
odd failure ...hhmmm...

**Comment by keradus** [comment]
7.2 is failing only when apply 7+ fixers...

File: dev-tools/composer.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "require": {
+        "php": "^7.0"
+    },
+    "require-dev": {
+        "maglnet/composer-require-checker": "^0.1.4",
```
Comment (SpacePossum) [inline dev-tools/composer.json]: 0.1.5 might be a better match as 0.1.4 is know to be bugged

Comment (keradus) [reply dev-tools/composer.json]: :+1:

File: dev-tools/build.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -e
+
+# ensure that deps will work on lowest supported PHP version
+composer config platform.php 2> /dev/null || composer config platform.php 5.6.0
```
Comment (julienfalque) [inline dev-tools/build.sh]: 5.3?

File: dev-tools/build.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+# install package deps without dev-deps / remove already installed dev-deps
+composer update --no-interaction --no-progress --no-dev --prefer-stable
+composer info -D | sort
+
+# install box2 globally
+composer global show kherge/box -q || composer global require --no-interaction --no-progress kherge/box:^2.7
+
+# build phar file
+php -d phar.readonly=false $(composer config home)/vendor/bin/box build
```
Comment (julienfalque) [inline dev-tools/build.sh]: Missing newline at end of file.

File: .travis.yml [hunk 4]
```diff
@@ -5,122 +5,131 @@ sudo: false
...
+                - ./check_trailing_spaces.sh || travis_terminate 1
+                - ./dev-tools/vendor/bin/composer-require-checker check composer.json --config-file=.composer-require-checker.json
+
+
+        - &STANDARD_TEST_JOB
+            stage: Test
+            php: 7.0
+            install:
+                # Composer: enforce given Symfony components version
+                - 'if [ "$SYMFONY_VERSION" != "" ]; then sed -i "s/\"symfony\/\([^\"]*\)\": \"^2[^\"]*\"/\"symfony\/\1\": \"$SYMFONY_VERSION\"/g" composer.json; fi'
```
Comment (julienfalque) [inline .travis.yml]: Where is `$SYMFONY_VERSION` defined?

File: dev-tools/build.sh (hunk 2 above)
Comment (keradus) [reply dev-tools/build.sh]: great catch !

File: .travis.yml (hunk 4 above)
Comment (keradus) [reply .travis.yml]: restored, it was on 2.2 line, now it's also on this PR

File: .travis.yml [hunk 5]
```diff
@@ -5,122 +5,132 @@ sudo: false
...
+            script:
+                - php php-cs-fixer fix --rules @PHP70Migration:risky,@PHP71Migration,native_function_invocation -q || travis_terminate 1
+                - vendor/bin/phpunit || travis_terminate 1
+                - git checkout . -q
+                - php php-cs-fixer --diff --dry-run -v fix
+
+        -
+            <<: *STANDARD_TEST_JOB
+            php: nightly
+            env: COMPOSER_FLAGS="--ignore-platform-reqs" SYMFONY_DEPRECATIONS_HELPER=weak PHP_CS_FIXER_IGNORE_ENV=1
```
[the remark this answers is not in the document]
Comment (keradus) [reply .travis.yml]: yes, we need just one build to run with legacy tokenizer, we could put it to any of jobs, I took the same as on master to decrease conflicts

File: .travis.yml [hunk 6]
```diff
@@ -5,122 +5,132 @@ sudo: false
...
+                - composer info -d dev-tools -D | sort
+
+                - travis_retry composer update $DEFAULT_COMPOSER_FLAGS $COMPOSER_FLAGS
+                - composer info -D | sort
+            script:
+                # @TODO remove at 3.0
+                - git archive -o /dev/null HEAD -v 2>&1 | grep tests | grep \.php | grep -v tests/Test/Assert/AssertTokensTrait.php | grep -v tests/Test/AbstractFixerTestCase.php | grep -v tests/Test/AbstractIntegrationTestCase.php | grep -v tests/Test/IntegrationCase.php | grep -v tests/Test/IntegrationCaseFactory.php && (echo "UNKNOWN FILES DETECTED" && travis_terminate 1) || echo "NO UNKNOWN FILES"
+
+                - ./check_trailing_spaces.sh || travis_terminate 1
+                - ./dev-tools/vendor/bin/composer-require-checker check composer.json --config-file=.composer-require-checker.json
```
[the remark this answers is not in the document]
Comment (keradus) [reply .travis.yml]: `phpmd` was never used for 2.2 branch and violations of `phpmd` were not fixed on 2.2 line, thus we cannot start using it now.

File: .travis.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (julienfalque) [reply .travis.yml]: Then one needs to be removed on upper branches, can you do it while merging?

[the remark this answers is not in the document]
Comment (keradus) [reply .travis.yml]: yes

File: .travis.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (keradus) [reply .travis.yml]: i just remind myself: they were not fixed as php 5.3 was too old to fix them

**Pull request closed without merging**